File: 8vim/src/main/java/inc/flide/vim8/structures/CharacterSet.java
package inc.flide.vim8.structures;

import java.util.ArrayList;
import java.util.List;

public class CharacterSet {

    private String lowerCaseCharacters;
    private String upperCaseCharacters;

    public CharacterSet() {
        lowerCaseCharacters = "";
        upperCaseCharacters = "";
    }

    public String getLowerCaseCharacters() {
        return lowerCaseCharacters;
    }

    public void setLowerCaseCharacters(String lowerCaseCharacters) {
        this.lowerCaseCharacters = lowerCaseCharacters;
    }

    public String getUpperCaseCharacters() {
        return upperCaseCharacters;
    }

    public void setUpperCaseCharacters(String upperCaseCharacters) {
        this.upperCaseCharacters = upperCaseCharacters;
    }
}

File: 8vim/src/main/kotlin/inc/flide/vim8/models/CharacterSet.kt
package inc.flide.vim8.models

import arrow.optics.optics

@optics
data class CharacterSet(
    val lowerCaseCharacters: String = "",
    val upperCaseCharacters: String = ""
) {
    companion object
}

fun CharacterSet.isNotEmpty(): Boolean =
    lowerCaseCharacters.isNotEmpty() || upperCaseCharacters.isNotEmpty()
